lua/nordic/colors/init.lua

```diff
@@ -18,6 +18,12 @@ local function build_palette()
     C.white0 = (options.reduced_blue and C.white0_reduce_blue) or C.white0_normal
 
     -- Modify the palette before generating colors.
+
+    -- This flag allows users to check if the extended palette can be used.
+    -- On the first run only the base palette is available.
+    C.extended = false
```

Comment:
`on_palette` is called twice, but one of those times all values in the extended palette are `nil`.

Say I want to make `bg_visual` a blend of `bg` & `orange.base` this will **panic:**


```lua
on_palette = function(palette)
    -- the if is needed because on_palette is called twice and one time palette.bg is nil which breaks blend
    palette.bg_visual = require("nordic.utils").blend(palette.orange.base, palette.bg, 0.15)
    return palette
end,
```
because `blend` is passed a `nil` value (`bg`)...

So you would have to check if a value in the extended palette is **not** `nil` before preforming you operation `extended` just provides a standard way of doing that:
```lua
on_palette = function(palette)
    -- the if is needed because on_palette is called twice and one time palette.bg is nil which breaks blend
    if palette.extended then
        palette.bg_visual = require("nordic.utils").blend(palette.orange.base, palette.bg, 0.15)
    end
    return palette
end,
```

I am not sure if its the best solution because it adds a fair bit of complexity,

For reference, here's the section I added to the readme for `on_palette`:
> An example of overriding colors in the base palette & extended palette:
> 
> ```lua
> require 'nordic' .setup {
>     on_palette = function(palette)
>         palette.black1 = "#BF616A"
>         palette.bg = palette.yellow.base
>         return palette
>     end,
> }
> ```
> 
> Internally the provided function is called twice, so you can safely use colors form the extended palette if you check that `palette.extended` is true.
> 
> Example of how to change the cursorline color to orange:
> 
> 
> ```lua
> require 'nordic' .setup {
>     on_palette = function(palette)
>         -- If the palette is extended, we can safely use palette.bg
>         if palette.extended then
>             palette.bg_visual = require("nordic.utils").blend(palette.orange.base, palette.bg, 0.2)
>         end
>         return palette
>     end,
> }
> ```

Reply:
What if we added a third callback?

Reply:
> What if we added a third callback?

That's the other good option. It's definitely simpler:
```lua
on_palette = function(palette)
    -- the if is needed because on_palette is called twice and one time palette.bg is nil which breaks blend
    if palette.extended then
        palette.bg_visual = require("nordic.utils").blend(palette.orange.base, palette.bg, 0.15)
    end
    return palette
end,
```
vs
```lua
after_palette = function(palette)
    palette.bg_visual = require("nordic.utils").blend(palette.orange.base, palette.bg, 0.15)
    return palette
end,
```
Another example:
```lua
on_palette = function(palette)
    local old_black = palette.black1
    palette.black1 = "#BF616A" -- red
    palette.bg = old_black -- black
    palette.fg = palette.yellow.base -- yellow
    palette.visual = palette.fg -- yellow
    palette.bg_float = palette.black1 -- red
    return palette
end,
```
vs
```lua
on_palette = function(palette)
    palette.old_black = palette.black1
    palette.black1 = "#BF616A" -- red
    return palette
end,
after_palette = function(palette)
    palette.bg = palette.old_black -- black
    palette.fg = palette.yellow.base -- yellow
    palette.visual = palette.fg -- yellow
    palette.bg_float = palette.black1 -- red
    return palette
end,
```
It's up to you; a third callback is simpler, and twice-called `on_palette` is generally shorter.

Reply:
Yeah, third callback is definitely the way to go/